lib/cocoapods/downloader/request.rb

```diff
@@ -21,7 +21,7 @@ class Request
 
       # @return [Hash<Symbol, String>] The download parameters for this request.
       #
-      attr_reader :params
+      attr_accessor :params
```

Comment:
i'd rather keep `Request` immutable and just construct a new instance instead of mutating?